### PR TITLE
Feature/project init name

### PR DESCRIPTION
**Pull request by m-mcgowan**
accepts a name for project init
adds (temporary) templates for project.ino and README.md
expands the templates when creating a new project

### CODE REVIEW

File: src/cmd/project_init.js [hunk 1]
```diff
@@ -65,49 +74,87 @@ export class ProjectInitCommandSite extends CommandSite {
...
 
+	static templateFile(name) {
+		return path.join(__dirname, 'templates', 'project', name);
+	}
+
+	static expandTemplate(fs, templateName, data) {
+		const readFile = promisify(fs.readFile);
+		return readFile(ProjectInitCommand.templateFile(templateName), 'utf-8')
+			.then(content => {
+				return require('underscore').template(content)(data);
```
Comment (monkbroc) [inline src/cmd/project_init.js]: Could you break `require('underscore')` to its own method named something like `templateEngine` or something to make the dependency more visible.

File: src/cmd/project_init.js [hunk 2]
```diff
@@ -65,49 +74,87 @@ export class ProjectInitCommandSite extends CommandSite {
...
 	 * @param {ProjectInitCommandSite} site external services.
 	 * @returns {Promise} To run the project initialization command.
 	 */
 	run(state, site) {
-		let directory, filesystem;
+		let directory, filesystem, name;
 
 		return Promise.resolve()
-			.then(() => Promise.resolve(site.directory()))
+		.then(() => Promise.resolve(site.name()))
```
Comment (monkbroc) [inline src/cmd/project_init.js]: It's not necessary to have extra `Promise.resolve` in these inner calls. Did you have a special reason for adding them?

File: src/cmd/project_init.js (hunk 1 above)
Comment (m-mcgowan) [inline src/cmd/project_init.js]: ok.

File: src/cmd/project_init.js (hunk 2 above)
Comment (m-mcgowan) [inline src/cmd/project_init.js]: yeah, totally agree it's not needed. I think it was just a habit to indicate that these site callbacks can return a value or a promise.

**Pull request merged**